spec/lib/binary-buffer-module.spec.js

```diff
@@ -0,0 +1,40 @@
+"use strict";
+
+describe("binaryBuffer", () => {
+    var binaryBuffer, copy;
+
+    beforeEach(() => {
+        binaryBuffer = require("../../lib/binary-buffer")();
+    });
+    [
+        "some string",
+        "",
+        new Buffer("some buffer"),
+        123
+    ].forEach((scenario) => {
+        describe(".toBuffer()", () => {
+            it("should convert a string to a buffer", () => {
+                scenario = binaryBuffer.toBuffer(scenario);
+                expect(scenario).toEqual(jasmine.any(Buffer));
+            });
+            it("should not alter the stringOrBuffer if it is already a buffer", () => {
+                copy = scenario;
+                scenario = binaryBuffer.toBuffer(scenario);
+                expect(scenario).toEqual(jasmine.any(Buffer));
+                expect(scenario).toBe(copy);
```

Comment:
It is interesting that this even works.  When copy is a number, why does `expect(scenario).toBe(copy)` pass?